Console/ReencryptColumn.php

```diff
@@ -0,0 +1,163 @@
+                'Whether to force this action to take effect'
+            ),
+            new InputArgument(
+                self::INPUT_KEY_TABLE,
+                null,
+                'The table containing the data to re-encrypt',
+                ''
+            ),
+            new InputArgument(
+                self::INPUT_KEY_IDENTIFIER,
+                null,
+                'The entity_id, row_id, or equivalent for the table',
+                ''
+            ),
+            new InputArgument(
+                self::INPUT_KEY_COLUMN,
+                null,
+                'The column that you want to re-encrypt',
+                ''
+            )
+        ];
+
+        $this->setName('gene:encryption-key-manager:reencrypt-column');
+        $this->setDescription('Re-encrypt a columns data with the latest key');
+        $this->setDefinition($options);
+
+        parent::configure();
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if (!$input->getOption(self::INPUT_KEY_FORCE)) {
+            $output->writeln('<info>Run with --force to make these changes, this will run in dry-run mode by default</info>');
+        }
+
+        try {
+            $keys = preg_split('/\s+/s', trim((string)$this->deploymentConfig->get('crypt/key')));
+            $latestKeyNumber = count($keys) - 1;
+            $output->writeln("The latest encryption key is number $latestKeyNumber, looking for old entries");
+
+            $table = $input->getArgument(self::INPUT_KEY_TABLE);
+            if (!strlen($table)) {
+                throw new \Exception('Provide a table name');
+            }
+            if (in_array($table, ['core_config_data', 'tfa_user_config'])) {
+                throw new \Exception('You cannot use this command for this table');
+            }
+            $identifier = $input->getArgument(self::INPUT_KEY_IDENTIFIER);
+            if (!strlen($identifier)) {
+                throw new \Exception('Provide an identifier');
+            }
+            $column = $input->getArgument(self::INPUT_KEY_COLUMN);
+            if (!strlen($column)) {
+                throw new \Exception('Provide an column');
+            }
+            $output->writeln("Looking for '$column' in '$table', identified by '$identifier'");
+
+            /**
+             * @see \Magento\Framework\Model\ResourceModel\Db\AbstractDb::_getLoadSelect()
+             */
+            $tableName = $this->resourceConnection->getTableName($table);
+            $connection = $this->resourceConnection->getConnection();
+            $field = $connection->quoteIdentifier(sprintf('%s.%s', $tableName, $column));
+
+            $select = $connection->select()
+                ->from($tableName, [$identifier, "$column"])
+                ->where("($field LIKE '_:_:____%' OR $field LIKE '__:_:____%')")
```

Comment:
Or to add additional condition here
```suggestion
                ->where("$field IS NOT NULL")
                ->where("($field LIKE '_:_:____%' OR $field LIKE '__:_:____%')")
```

Reply:
Also the filtering for nully values on non-indexed tables is longer than selecting the data and filtering in PHP land 👍